### PR TITLE
Add user documentation for the BigQuery source.

### PR DESCRIPTION
**Pull request by danpi**
### Motivation

- Add user documentation.

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: docs/google-bigquery-source.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+| `credentialJsonString`       | String | No       | "" (empty string) | The authentication JSON key. Set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to the path of the JSON file that contains your service account key when the `credentialJsonString` is set to an empty string. For details, see the [Google documentation](https://cloud.google.com/bigquery/docs/quickstarts/quickstart-client-libraries#before-you-begin). |
+| `queueSize`                  | int    | No       | "" (empty string) | The buffer queue size of the source. It is used for store records before they are sent to Pulsar topics. By default, it is set to 10000.                                                                                                                                                                                                                                |
```
Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:55]:
```suggestion
| `queueSize`                  | int    | No       | "" (empty string) | The buffer queue size of the source. It is used for storing records before they are sent to Pulsar topics. By default, it is set to `10000`.                                                                                                                                                                                                                                |
```

File: docs/google-bigquery-source.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+| `snapshotTime`               | long   | No       | -1                | The snapshot time of the table. If not set, interpreted as now.                                                                                                                                                                                                                                                                                                         |                                                                                                                                                                                                                                                                                                                      |
```
Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:56]:
```suggestion
| `snapshotTime`               | long   | No       | -1                | The snapshot time of the table. If it is not set, it is interpreted as now.                                                                                                                                                                                                                                                                                                         |                                                                                                                                                                                                                                                                                                                      |
```

File: docs/google-bigquery-source.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+| `sql`                        | String | No       | "" (empty string) | The SQL query on BigQuery and save the computed result in a temporary table. The temporary table has a configurable expiration time,and the source-connector automatically deletes the temporary table when the data transfer is completely. The projectId and datasetName use values in config,and the tableName will be generated by UUID.                            |                                                                                                                                                                                                                                                                                                                     |
```
Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:57]:
```suggestion
| `sql`                        | String | No       | "" (empty string) | The SQL query on BigQuery. The computed result is saved in a temporary table. The temporary table has a configurable expiration time, and the BigQuery source connector automatically deletes the temporary table when the data is transferred completely. The `projectId` and `datasetName` gets values from the configuration file, and the `tableName` is generated by UUID.                            |                                                                                                                                                                                                                                                                                                                     |
```

File: docs/google-bigquery-source.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+| `expirationTimeInMinutes`    | int    | No       | 1440              | The expiration time in minutes until the table is expired and auto-deleted.                                                                                                                                                                                                                                                                                             |
+| `maxParallelism`             | int    | No       | 1                 | The max parallelism for reading. The maximal number of partitions to split the data into. Actual number may be less if BigQuery deems the data small enough.                                                                                                                                                                                                            |
```
Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:59]:
```suggestion
| `maxParallelism`             | int    | No       | 1                 | The maximum parallelism for reading. In fact, the number may be less if the BigQuery source connector deems the data small enough.                                                                                                                                                                                                            |
```

Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:59]:
```suggestion
| `maxParallelism`             | int    | No       | 1                 | The maximum parallelism for reading. In actual, the number may be less if the BigQuery source connector deems the data small enough.                                                                                                                                                                                                            |
```

File: docs/google-bigquery-source.md [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+| `filters`                    | String | No       | "" (empty string) | A list of clauses that can filter the result of the table.                                                                                                                                                                                                                                                                                                              |
+| `checkpointIntervalSeconds`  | int    | No       | 60                | The checkpoint interval(in units of seconds). By default, it is set to 60s.                                                                                                                                                                                                                                                                                             |
```
Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:62]:
```suggestion
| `checkpointIntervalSeconds`  | int    | No       | 60                | The checkpoint interval (in units of seconds). By default, it is set to 60s.                                                                                                                                                                                                                                                                                             |
```

File: docs/google-bigquery-source.md [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+            "datasetName": "babynames",
+            "tableName": "names2021",
+            "credentialJsonString": "SECRETS"
+        }
+    }
+    ```
+
+## Work with Function Mesh
+
+You can create a [CustomResourceDefinitions (CRD)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to create a Google Cloud BigQuery source connector. Using CRD makes Function Mesh naturally integrate with the Kubernetes ecosystem. For more information about Pulsar source CRD configurations, see [source CRD configurations](https://functionmesh.io/docs/connectors/io-crd-config/source-crd-config).
```
Comment (Huanli-Meng) [inline docs/google-bigquery-source.md:138]:
```suggestion
You can create a [CustomResourceDefinitions (CRD)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to create a Google Cloud BigQuery source connector. Using CRD makes Function Mesh naturally integrate with the Kubernetes ecosystem. For more information about Pulsar IO source CRD configurations, see [source CRD configurations](https://functionmesh.io/docs/connectors/io-crd-config/source-crd-config).
```

**Review by Huanli-Meng — COMMENTED** [review]
Leave some comments. PTAL

**Comment by danpi** [comment]
> 

Thanks a lot for the suggestions, already applied.

**Pull request merged**